daml/ContingentClaims/ObservableFunctions.daml

```diff
@@ -9,32 +9,33 @@ module ContingentClaims.ObservableFunctions where
 
 -- TODO: rename this module to something more appropriate
 
-import Daml.Control.Category (Category, id)
-import Prelude (Applicative, Additive, Multiplicative, Divisible, Ord, aunit, munit, liftA2)
-import Prelude qualified ((+),negate,(*),(/),(<=),(==),pure)
+import Daml.Control.Arrow (Arrow, arr)
+import Daml.Data.Functor.Contravariant
+import Prelude (Applicative, Additive, Multiplicative, Divisible, liftA2, undefined, Text, const, (<=), (.))
+import Prelude qualified ((+), negate, (*), (^), (/), pure, aunit, munit)
 import ContingentClaims.Observable
 
 -- Default instances for applicative functors
 
-instance forall f t . Category f => TimeF f t where 
-  time = id
-
-instance forall f t a . Applicative (f t) => PointF f t a where
+instance forall f t . (Arrow f, Applicative (f t)) => Point f t Text where
   pure = Prelude.pure
+  observe key = arr (const key)
 
-instance forall f t a . (Additive a, Multiplicative a, Divisible a, Applicative (f t)) => RingF f t a where
-  zero = Prelude.pure aunit
-  one = Prelude.pure munit
+instance (Additive a, Applicative (f t)) => Additive (f t a) where
+  aunit = Prelude.pure Prelude.aunit
   (+) = liftA2 (Prelude.+)
-  negate = fmap (Prelude.negate)
+  negate = fmap Prelude.negate
+
+instance (Multiplicative a, Applicative (f t)) => Multiplicative (f t a) where
+  munit = Prelude.pure Prelude.munit
   (*) = liftA2 (Prelude.*)
-  (/) = liftA2 (Prelude./)
+  (^) = undefined
 
```

Comment:
Why is this undefined, is there no instance to lift?

Reply:
Good catch; I was just being lazy.

I've also fixed this for `Observation`, but only for powers 0,1, 2. Otherwise, we run the risk of creating super nested trees (as I've just implemented as unfolding `Mul` _n_ times). Not sure if that will be an issue.